### PR TITLE
sample

### PR DESCRIPTION
**Pull request by natmaka**
HI,

This shows which sort of potential problems are detected by shellcheck 

I did not test, please beware

Take care,

### CODE REVIEW

**Comment by hinto-janai** [comment]
Hi, I don't usually use shellcheck but some of these changes are good

Can you merge into `dev` instead of `main`?

**Comment by natmaka** [comment]
Hi,

You may disable shellcheck's hints which seem useless/bad to you, keeping only those you like.  Given the nature of your code it will be in my opinion better to be very strict.

File: src/func/backup [hunk 1]
```diff
@@ -12,10 +12,9 @@ backup_Wallets()
 
 	# PASSWORD-BASED AES256 ENCRYPTION (output file: backup.tar.gpg)
 	$bblue; echo "Backing up wallets..." ;$off
-	cd "$dotMoneroBash"
+	cd "$dotMoneroBash" || die "Cannot cd to directory $dotMoneroBash"
 	tar -cf "backup.tar" "wallets" --transform 's/wallets/backup/'
-	gpg --no-symkey-cache --cipher-algo AES256 -c "backup.tar"
-	if [[ $? != 0 ]]; then
+ 	if gpg --no-symkey-cache --cipher-algo AES256 -c "backup.tar" ; then
```
Comment (hinto-janai) [inline src/func/backup:17]: should be `if ! gpg --no-symkey-cache --cipher-algo AES256 -c "backup.tar"; then`

File: src/func/seed [hunk 2]
```diff
@@ -19617,16 +19617,16 @@ esac
 seed_Rng()
 {
 	# 64 bytes/512 bits of /dev/random fed into sha512sum, only integers printed
-	sha="$(cat /dev/random | head -c 64 | sha512sum | sed 's/[^0-9]*//g')"
+	sha="$(head -c 64 /dev/random | sha512sum | sed 's/[^0-9]*//g')"
 
 	# only print first 4 integers
-	number="$(printf ${sha:0:4})"
+	number=$(printf "%s " "${sha:0:4}")
```
Comment (hinto-janai) [inline src/func/seed:19623]: I think when printing a single regular string `printf "text"` is fine, could you revert these?

File: src/func/seed [hunk 3]
```diff
@@ -19645,30 +19645,35 @@ seed_Generate()
...
 	for i in $list ;do
-		printf "${i:0:3}"
+		printf "%s" "${i:0:3}"
 	done
 }
 
 # seed = seperated by spaces, list = seperated by newlines
-local seed="$(seed_Generate)"
-local list="$(echo "$seed" | sed "s@ @\n@g")"
+local seed
```
Comment (hinto-janai) [inline src/func/seed:19665]: you don't need to initialize and define a variable on separate lines

**Review by hinto-janai — CHANGES_REQUESTED** [review]
Everything else is good :D

**Pull request closed without merging**